apps/web/lib/hooks/usePublicPage.ts

```diff
@@ -1,9 +1,12 @@
-import { usePathname } from "next/navigation";
+import { usePathname, useSearchParams } from "next/navigation";
 
 export default function usePublicPage() {
   const pathname = usePathname();
-  const isPublicPage = ["/[user]", "/booking", "/cancel", "/reschedule"].find((route) =>
-    pathname?.startsWith(route)
-  );
-  return isPublicPage;
+  const isPublicPage = ["/booking", "/cancel", "/reschedule"].some((route) => pathname?.startsWith(route));
```

Comment:
/[user] url wasn't executing properly, so we now look for searchParams with the new next/navigation

Reply:
I think now we can move to using `isBookingPage` Component prop(which wasn't there earlier) as that is more reliable 
![image](https://github.com/calcom/cal.com/assets/1780212/45a00858-5f4d-40f2-ab97-5d8850881e2f)

Reply:
Should we rename? isBookingPage just tells that its a booking page but it will be better if we use isPublicPage? or auth = true?
https://next-auth.js.org/getting-started/client#custom-client-session-handling

Reply:
The purpose of isPublicPage was to disable intercom on booking pages that are public. So, I guess we would be fine renaming it to useIsBookingPage

Reply:
Agree